pointcloud/basic_octree.go

```diff
@@ -129,3 +131,55 @@ func (octree *BasicOctree) Iterate(numBatches, currentBatch int, fn func(p r3.Ve
 func (octree *BasicOctree) MetaData() MetaData {
 	return octree.meta
 }
+
+// CollidesWithGeometry will return whether a given geometry is in collision with a given point.
+func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, threshold int, buffer float64) (bool, error) {
+	if octree.MaxVal() < threshold {
+		return false, nil
+	}
+	ocbox, err := spatialmath.NewBox(
+		spatialmath.NewPoseFromPoint(octree.center),
+		r3.Vector{octree.sideLength + buffer, octree.sideLength + buffer, octree.sideLength + buffer},
+		"",
+	)
+	if err != nil {
+		return false, err
+	}
+
+	// Check whether our geom collides with the area represented by the octree. If false,
+	collide, err := geom.CollidesWith(ocbox)
+	if err != nil {
+		return false, err
+	}
+	if !collide {
+		return false, nil
+	}
+
+	if octree.node.point != nil {
```

Comment:
see below comment but this check might not be necessary (although still might be a good idea) if this is only being done in a filledLeafNode as they should ALWAY have a point